### PR TITLE
Implement `/connect`

### PR DESCRIPTION
**Pull request by glsubri**

### CODE REVIEW

**Comment by glsubri** [comment]
Note that this will add the test from PR 105.
It will close issue #109

File: src/main/java/ch/heigvd/pro/b04/sessions/Session.java [hunk 1]
```diff
@@ -4,40 +4,74 @@
...
+@AllArgsConstructor
+@Builder
+@Entity
+@EqualsAndHashCode
 public class Session {
+  private static final int CODE_BOUNDARY = 0xFFFF;
+
+  public enum SessionState {
+    OPEN, CLOSED_TO_NEW_ONES, CLOSED
+  }
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/sessions/Session.java]: We might want to remove the `Session` prefix from `SessionState`, so we can us it by writing `Session.State`.

File: src/main/java/ch/heigvd/pro/b04/sessions/Session.java [hunk 2]
```diff
@@ -4,40 +4,74 @@
...
 
 
-@Entity
 @NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@EqualsAndHashCode
 public class Session {
+  private static final int CODE_BOUNDARY = 0xFFFF;
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/sessions/Session.java]: According to the [documentation](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html#nextInt-int-), the parameter of `Random.nextInt(something)` is exclusive.

File: src/main/java/ch/heigvd/pro/b04/participants/Participant.java [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
   @EmbeddedId
   private ParticipantIdentifier idParticipant;
 
-  private String username;
+  @Column(unique = true)
+  @Getter
+  private String token;
 
-  public Participant(long id, String username) {
+  public Participant(long id, UserToken token) {
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/participants/Participant.java]: When using this constructor, don't we lose information related to the `Session` to which the participant belongs ?

File: src/test/java/ch/heigvd/pro/b04/session/SessionControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package ch.heigvd.pro.b04.session;
```
Comment (alexandrepiveteau) [inline src/test/java/ch/heigvd/pro/b04/session/SessionControllerTest.java]: Maybe not the right package.

File: src/test/java/ch/heigvd/pro/b04/session/SessionCodeTest.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package ch.heigvd.pro.b04.session;
```
Comment (alexandrepiveteau) [inline src/test/java/ch/heigvd/pro/b04/session/SessionCodeTest.java]: Maybe not the right package.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

**Review by alexandrepiveteau — APPROVED** [review]
LGTM

**Pull request merged**